src/service.c

```diff
@@ -298,6 +291,12 @@ static GVariant* create_slotstatus_array(void)
 		g_clear_error(&ierror);
 	}
 
+	res = determine_boot_states(&ierror);
```

Comment:
Do we need to call the bootloader interface every time someone calls the dbus method? Currently we even call the bootloader tool multiple times (once for each slot with a bootname).